source/common/ssl/context_config_impl.cc

```diff
@@ -41,13 +41,27 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::CommonTlsContext& con
           RepeatedPtrUtil::join(config.tls_params().cipher_suites(), ":"), DEFAULT_CIPHER_SUITES)),
       ecdh_curves_(StringUtil::nonEmptyStringOrDefault(
           RepeatedPtrUtil::join(config.tls_params().ecdh_curves(), ":"), DEFAULT_ECDH_CURVES)),
-      ca_cert_file_(config.validation_context().trusted_ca().filename()),
-      cert_chain_file_(config.tls_certificates().empty()
-                           ? ""
-                           : config.tls_certificates()[0].certificate_chain().filename()),
-      private_key_file_(config.tls_certificates().empty()
-                            ? ""
-                            : config.tls_certificates()[0].private_key().filename()),
+      ca_cert_(readDataSource(config.validation_context().trusted_ca(), true)),
+      ca_cert_path_(config.validation_context().trusted_ca().specifier_case() ==
+                            envoy::api::v2::DataSource::kFilename
+                        ? config.validation_context().trusted_ca().filename()
+                        : ""),
+      cert_chain_(config.tls_certificates().empty()
+                      ? ""
+                      : readDataSource(config.tls_certificates()[0].certificate_chain(), true)),
```

Comment:
Is there a reasonable way to make the initializors here a bit cleaner, e.g. with anonymous namespace functions etc.

Reply:
While I agree that this isn't the prettiest code, I'm with Greg on this, it's much more readable here.